src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -187,3 +195,57 @@ def test_unordered_list_match(inputs, outputs):
         assert all(input in outputs for input in inputs)
     except KeyError as exception:
         raise AssertionError("lists do not match") from exception
+
+
+def transform_model(
+    input_model, output_model, property_transform_keys, property_transform
+):
+    for prop in property_transform_keys:
+        document_input, _path_input, _parent_input = traverse(
+            input_model, list(prop)[1:]
+        )
+        document_output, _path_output, _parent_output = traverse(
+            output_model, list(prop)[1:]
+        )
+        if document_input != document_output:
+            transformed_property = transform(prop, input_model, property_transform)
+            update_transformed_property(prop, transformed_property, input_model)
+
+
+def transform(property_path, input_model, property_transform):
+    path = set_to_string(property_path)
+    file = open("./transformation.js", "a")
+    file.write("var jsonata =require('jsonata')\n")
+    file.write("var model = {0};\n".format(input_model))
+    property_transform_value = property_transform[path].replace('"', '\\"')
+    file.write('var expression = jsonata("{0}");\n'.format(property_transform_value))
+    file.write("var result = expression.evaluate(model);\n")
+    file.write("console.log(result);\n")
+    file.close()
+    return execute_js()
+
+
+def set_to_string(property_path):
+    path_string = ""
+    for path in property_path:
+        path_string += "/" + path
+
+    return path_string
+
+
+def update_transformed_property(property_path, transformed_property, input_model):
+    try:
+        _prop, resolved_path, parent = traverse(input_model, list(property_path)[1:])
+    except LookupError:
+        LOG.debug("Override failed.\nPath %s\nDocument %s", property_path, input_model)
+        LOG.warning("Override with path %s not found, skipping", property_path)
+    else:
+        key = resolved_path[-1]
+        parent[key] = transformed_property
+
+
+def execute_js():
+    subprocess.getoutput("npm install jsonata")
+    jsonata_output = subprocess.getoutput("node transformation.js")
```

Comment:
[Still think we should re-consider pulling in another programming language just for this](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/pull/111#discussion_r555242773)

Reply:
Also could look into some evaluation lib: https://github.com/sony/v8eval

Reply:
Still think we should re-consider pulling in another programming language just for this
> dont think this is a possibility now. We have had design conversations on this and the backend service is already implemented for jsonata

Reply:
Also could look into some evaluation lib: https://github.com/sony/v8eval\
> I looked into adding both v8eval & js2py. Both are running into issues when fetching jsonata into code. Right now, subprocess is the easiest way to include jsonata code. The other option is to write our own python wrapper for jsonata. We could consider doing that for the long run